operator/validator/controller.go

```diff
@@ -889,20 +888,11 @@ func (c *controller) onShareStop(pubKey spectypes.ValidatorPK) {
 	// stop instance
 	v.Stop()
 	c.logger.Debug("validator was stopped", fields.PubKey(pubKey[:]))
-	vc, ok := c.validatorsMap.GetCommittee(v.Share().CommitteeID())
-	if ok {
+
+	if !c.validatorsMap.UpdateCommitteeAtomic(v.Share().CommitteeID(), func(vc *validator.Committee) {
 		vc.RemoveShare(v.Share().Share.ValidatorIndex)
-		if len(vc.Shares) == 0 {
-			deletedCommittee := c.validatorsMap.RemoveCommittee(v.Share().CommitteeID())
-			if deletedCommittee == nil {
-				c.logger.Warn("could not find committee to remove on no validators",
-					fields.CommitteeID(v.Share().CommitteeID()),
-					fields.PubKey(pubKey[:]),
-				)
-				return
-			}
-			deletedCommittee.Stop()
-		}
+	}) {
+		c.logger.Warn("committee not found", fields.PubKey(pubKey[:]))
```

Comment:
Similar to https://github.com/ssvlabs/ssv/pull/1760/files#r1840850336, probably want to log this as `Error` (not `Warn`) ? Because if committee is not found here - looks like something is wrong with the way we manage it's lifecycle, and we'd need to address it (warnings in logs are pretty easy to miss, compared to errors).

Reply:
As I remember in past conversations it was decided not to change the log level and keep it as it was previously (line 898)